### ISSUE TITLE
SSL error connection: certificate has expired

### ISSUE THREAD

**Issue by sir-marv** [LukeSmithxyz/emailwiz#101]
I am using emailwiz along with muttwizard and I've had this problem a few times. After about a month, my mw -Y gives me the error " SSL error connection mail.hostname.com: certificate has expired". I run certbot to renew the cert (even though its valid for another 2 months).

The only way I've been able to get it to work is to rebuild from scratch the entire email server and setup a new a fresh mw acocunt.

**Comment by sir-marv** [comment]
Rerunning emailwiz.sh and using the new TXT record seems to fix this issue. Any ideas why this happens every few weeks to me? This is the third time it has happened.

**Comment by DAMO238** [comment]
I have the same issue. The interesting thing is that if you go to subdomain `mail.example.com`, it shows the renewed certificate. Checking the dovecot config, it is pointing to the new certificate. Postfix does not seem to have this issue. Simply reloading the dovecot service fixes this issue. If you wanted to automate this, you could add this reload into your cron job for renewing certificates.

TL;DR: Run `service dovecot reload`

**Comment by LukeSmithxyz** [comment]
Yeah, you might just want to have your dovecot service reload whether your certs are renewed checked.

**Comment by lapwat** [comment]
What are the exact steps to renew the certificate?

After running: `service dovecot reload`

The certificate is still expired.

https://ssl-tools.net/mailservers

**Comment by mariansam** [comment]
If anyone finds this, certbot automatically runs successful renewal hooks located in `/etc/letsencrypt/renewal-hooks/deploy`, so you can create the file

`/etc/letsencrypt/renewal-hooks/deploy/50-dovecot-reload.sh`:
```sh
#!/bin/sh
echo RENEWED_DOMAINS = $RENEWED_DOMAINS
for domain in $RENEWED_DOMAINS
do
        echo domain = $domain
        if [ "$domain" = mail.mariansam.eu ]
        then
                echo yeye reloading dovecot
                doveadm reload >/dev/null
        fi
done
```
(of course change `mail.mariansam.eu` for your email domain; and you can remove the `echo` commands)

and make it executable by `chmod +x 50-dovecot-reload.sh`.

An example output after running `certbot renew --force-renewal --cert-name mariansam.eu`:

<details>
<summary>Click to open</summary>

```
Saving debug log to /var/log/letsencrypt/letsencrypt.log

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Processing /etc/letsencrypt/renewal/mariansam.eu.conf
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Plugins selected: Authenticator nginx, Installer nginx
Renewing an existing certificate
Running deploy-hook command: /etc/letsencrypt/renewal-hooks/deploy/50-dovecot-reload.sh
Output from 50-dovecot-reload.sh:
RENEWED_DOMAINS = mariansam.eu mail.mariansam.eu www.mail.mariansam.eu www.mariansam.eu
domain = mariansam.eu
domain = mail.mariansam.eu
yeye reloading dovecot
domain = www.mail.mariansam.eu
domain = www.mariansam.eu


- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
new certificate deployed with reload of nginx server; fullchain is
/etc/letsencrypt/live/mariansam.eu/fullchain.pem
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

Congratulations, all renewals succeeded. The following certs have been renewed:
  /etc/letsencrypt/live/mariansam.eu/fullchain.pem (success)
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
```

</details>

The renewal is done automatically by a systemd timer (see `systemctl list-timers` and files `/lib/systemd/system/certbot.timer` and `/lib/systemd/system/certbot.service`). I'm on Debian 10 Buster. This is already working for me for a long time, but my certificates don't expire every day, so I can test the Dovecot reloading right now, but I believe it must be working, if the manual command above worked.

I think Postfix reloads the certificates automatically as they get renewed, my emails were arriving even when K-9 Mail and neomutt couldn't connect because of the old certificate.

**Comment by lapwat** [comment]
After investigation it looks like you need to rerun the wizard and reload nginx, not dovecot.

```
./emailwiz.sh
service nginx reload
```

And it should be good

**Comment by mariansam** [comment]
@lapwat you mean I should do this every time the certificate expires? Or will your solution fix the problem forever? I don't see why running the script again should solve the issue. I set up my mail server using emailwiz almost two years ago and I'm pretty scared to run the script again.

**Comment by lapwat** [comment]
I run the wizard everytime my certificate expires (once every 6 months) and it generates a new certificate. It does not reset your emails and email accounts.

**Comment by LukeSmithxyz** [comment]
You absolutely do not need to rerun this script. Just reload/restart postfix/dovecot. Only if you change the location of your certificate would you need to change the directory/file of the certificate in the configuration for both. (I suppose that could happen on machines where you are adding new domains to one large certificate, perhaps.)

**Comment by mariansam** [comment]
Luke, how do you handle this? Do you manually `systemctl reload dovecot`/`devadm reload` every three months? Or do you have a similar post-renewal hook set up? Don't you think this should be part of this script?
